snapcraft/meta/manifest.py

```diff
@@ -103,7 +103,7 @@ def write(
         version=version,
         summary=project.summary,  # type: ignore
         description=project.description,  # type: ignore
-        base=cast(str, project.base),
+        base=project.base,
```

Comment:
can you add above 
```python
base: Optional[str] = None
if project.base is not None:
    base = cast(str, project.base)
```

I am not sure why @cmatsuoka added the cast now, so maybe no action is required.

Reply:
I see the error of my comment now :facepalm:

Reply:
With the conditional casting is not required.

Reply:
Yep. `Project.base` is an `Optional[str]`, so changing `Manifest.base` to the same type removed the need to cast (and seems to be the correct data type anyway).

As the yaml is generated with `exclude_none=True`, the resulting manifest won't have a base attribute if `Project.base is None`. That seems to match the behaviour for snapcraft_legacy w.r.t. this attribute.